rosidl_typesupport_introspection_tests/test/test_typesupport_c.cpp

```diff
@@ -0,0 +1,2215 @@
+    all_messages_member_counts.insert(
+      interfaces_message_member_counts.begin(),
+      interfaces_message_member_counts.end());
+    for (const auto & service_type: services) {
+      service_messages.push_back(service_type + "_Request");
+      service_messages.push_back(service_type + "_Response");
+      all_services.push_back(service_type);
+
+      all_messages_member_counts[service_type + "_Request"] =
+        services_message_member_counts[service_type + "_Request"];
+      all_messages_member_counts[service_type + "_Response"] =
+        services_message_member_counts[service_type + "_Response"];
+    }
+    for (const auto & action_type: actions) {
+      action_messages.push_back(action_type + "_Feedback");
+      action_messages.push_back(action_type + "_FeedbackMessage");
+      action_messages.push_back(action_type + "_GetResult_Request");
+      action_messages.push_back(action_type + "_GetResult_Response");
+      action_messages.push_back(action_type + "_Goal");
+      action_messages.push_back(action_type + "_Result");
+      action_messages.push_back(action_type + "_SendGoal_Request");
+      action_messages.push_back(action_type + "_SendGoal_Response");
+
+      action_services.push_back(action_type + "_GetResult");
+      action_services.push_back(action_type + "_SendGoal");
+
+      all_messages_member_counts[action_type + "_Feedback"] =
+        action_message_member_counts[action_type + "_Feedback"];
+      all_messages_member_counts[action_type + "_FeedbackMessage"] =
+        action_message_member_counts[action_type + "_FeedbackMessage"];
+      all_messages_member_counts[action_type + "_GetResult_Request"] =
+        action_message_member_counts[action_type + "_GetResult_Request"];
+      all_messages_member_counts[action_type + "_GetResult_Response"] =
+        action_message_member_counts[action_type + "_GetResult_Response"];
+      all_messages_member_counts[action_type + "_Goal"] =
+        action_message_member_counts[action_type + "_Goal"];
+      all_messages_member_counts[action_type + "_Result"] =
+        action_message_member_counts[action_type + "_Result"];
+      all_messages_member_counts[action_type + "_SendGoal_Request"] =
+        action_message_member_counts[action_type + "_SendGoal_Request"];
+      all_messages_member_counts[action_type + "_SendGoal_Response"] =
+        action_message_member_counts[action_type + "_SendGoal_Response"];
+    }
+    all_messages = interfaces;
+    all_messages.insert(
+      all_messages.end(),
+      service_messages.begin(),
+      service_messages.end());
+    all_messages.insert(
+      all_messages.end(),
+      action_messages.begin(),
+      action_messages.end());
+    all_services.insert(
+      all_services.end(),
+      action_services.begin(),
+      action_services.end());
+
+    auto typesupport_lib_name = "lib" + ns + "__rosidl_typesupport_introspection_c.so";
+    ts_lib = dlopen(typesupport_lib_name.c_str(), RTLD_LAZY);
```

Comment:
@gbiggs using [`rcpputils::SharedLibrary`](https://github.com/ros2/rcpputils/blob/master/include/rcpputils/shared_library.hpp) may be best.

Reply:
That's a good idea. The C-language tests don't have to avoid C++, so using an existing utility for managing the shared library would make them more robust.